Assignment.Tests/PingProcessTests.cs

```diff
@@ -0,0 +1,262 @@
+        var outputAction = new Action<string?>(input => stringBuilder.AppendLine(input));
+        var errorAction = new Action<string?>(input => Console.WriteLine($"Error: {input}"));
+        
+        // Act
+        int exitCode = await Sut.RunLongRunningAsync(startInfo,outputAction, errorAction, default);
+
+        // Assert
+        AssertValidPingOutput(exitCode, stringBuilder.ToString());
+    }
+
+    // Not sure why this method is here so I'm commenting it out for the time being
+    //[TestMethod]
+    //public void StringBuilderAppendLine_InParallel_IsNotThreadSafe()
+    //{
+    //    IEnumerable<int> numbers = Enumerable.Range(0, short.MaxValue);
+    //    StringBuilder stringBuilder = new();
+    //    numbers.AsParallel().ForAll(item => stringBuilder.AppendLine(""));
+    //    int lineCount = stringBuilder.ToString().Split(Environment.NewLine).Length;
+    //    Assert.AreNotEqual(lineCount, numbers.Count()+1);
+    //}
+    //4
+    [TestMethod]
+     public async Task RunAsync_ReturnsCorrectResult()
+    {
+        // Arrange
+        CancellationToken cancellationToken = default;
+        
+        var hostNamesOrAddresses = new List<string> { $"{PingParameter} 4 localhost", $"{PingParameter} 4 localhost", $"{PingParameter} 4 localhost" };
+
+        //Act
+        PingResult result = await Sut.RunAsync(hostNamesOrAddresses, cancellationToken);
+
+        //Assert
+        AssertValidPingOutput(result);
+
+    }
+    //4
+    [TestMethod]
+    [ExpectedException(typeof(TaskCanceledException))]
+    public void RunAsync_ParallelQuery_WithCancellation()
+    {
+        // Arrange
+        CancellationTokenSource cancellationTokenSource = new();
+        cancellationTokenSource.Cancel();
+        var hostNamesOrAddresses = new List<string> { $"{PingParameter} 4 localhost", $"{PingParameter} 4 localhost", $"{PingParameter} 4 localhost", $"{PingParameter} 4 localhost" };
+
+        //Act
+        Task<PingResult> result = Sut.RunAsync(hostNamesOrAddresses, cancellationTokenSource.Token);
+
+        //Assert
+        try
+        {
+            result.Wait();
+        }
+        catch (AggregateException aggregateException)
+        {
+            foreach (Exception ex in aggregateException.InnerExceptions)
+            {
+                if (ex is TaskCanceledException)
+                {
+                    ExceptionDispatchInfo.Capture(ex).Throw();
+                }
+            }
+        }
+
+    }
+
+    // Windows version:
+    readonly string WindowsPingOutputLikeExpression = @"
+Pinging * with 32 bytes of data:
+Reply from ::1: time<*
+Reply from ::1: time<*
+Reply from ::1: time<*
+Reply from ::1: time<*
+
+Ping statistics for ::1:
+    Packets: Sent = *, Received = *, Lost = 0 (0% loss),
+Approximate round trip times in milli-seconds:
+    Minimum = *ms, Maximum = *ms, Average = *ms
+".Trim();
+
+    // Unix version:
+    readonly string UnixPingOutputLikeExpression = @"

```

Comment:
I really can't tell why your UnixPingOutputLikeExpression isnt work, it looks identical in the workflow to ours.
This is what we did and it worked:

```suggestion
    readonly string PingOutputLikeExpression = @"
PING * 56 data bytes
64 bytes from * (::1): icmp_seq=* ttl=* time=* ms
64 bytes from * (::1): icmp_seq=* ttl=* time=* ms
64 bytes from * (::1): icmp_seq=* ttl=* time=* ms
64 bytes from * (::1): icmp_seq=* ttl=* time=* ms
--- * ping statistics ---
* packets transmitted, * received, *% packet loss, time *ms
rtt min/avg/max/mdev = */*/*/* ms
".Trim();
```

Reply:
Thank you for the suggestion! I figured out it was the line endings. I had to normalize them I guess.